react/src/components/ComputeSessionNodeItems/TerminateSessionModal.tsx

```diff
@@ -250,54 +256,57 @@ const TerminateSessionModal: React.FC<TerminateSessionModalProps> = ({
       centered
       title={t('session.TerminateSession')}
       open={openTerminateModal}
-      confirmLoading={terminateMutation.isPending}
+      confirmLoading={pendingCount > 0}
       onOk={() => {
-        if (sessions[0]?.row_id) {
-          const session = sessions[0];
-          terminateMutation
-            .mutateAsync(session)
-            .then(() => {
-              setIsForce(false);
-              onRequestClose(true);
-            })
-            .catch((err) => {
-              upsertNotification({
-                message: painKiller.relieve(err?.title),
-                description: err?.message,
-                open: true,
-              });
-            })
-            .finally(() => {
-              // TODO: remove below code after session list migration to React
-              const event = new CustomEvent(
-                'backend-ai-session-list-refreshed',
-                {
-                  detail: 'running',
-                },
-              );
-              document.dispatchEvent(event);
+        const promises = _.map(
+          filterEmptyItem(_.castArray(sessions)),
+          (session) => {
+            return terminiateSession(session)
+              .catch((err) => {
+                upsertNotification({
+                  message: painKiller.relieve(err?.title),
+                  description: err?.message,
+                  open: true,
+                });
+              })
```

Comment:
If you delete multiple sessions, you'll get the same error message multiple times. How about setting a key for the notification (when deleting multiple pending sessions together)?

Reply:
Should we use error.message instead of err.description when displaying an error notification? The difference is that message contains the status code.

Reply:
1. Currently, it's the only way to let the user know which one has failed. Let's improve how we display errors in multiple mutations in the near future.
2. Yes, it's better to use `description`. Thank you. Also, I moved the code to dispatch event after all promises are settled.